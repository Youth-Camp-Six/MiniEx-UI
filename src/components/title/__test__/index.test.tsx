/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Title } from '..';

// 测试用例编写，此为伪代码
describe('title', () => {
  it('title snapshot test', () => {
    const { container } = render(
      <div>
        <Title size='s'>标题组件 s</Title>
        <Title size='l'>标题组件 l</Title>
        <Title size='m'>标题组件 m</Title>
        <Title size='xl'>标题组件 xl</Title>
        <Title size='xxl'>标题组件 xxl</Title>
      </div>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('test default title', () => {
    const { container } = render(<Title />);
    const fontSize = (container.firstChild as HTMLElement).style?.fontSize;
    expect(fontSize).toBe('16px');
    expect(container.firstChild).toHaveClass('mi-title');
  });

  it('test xxl title font-size', () => {
    const { container } = render(<Title size='xxl' />);
    const fontSize = (container.firstChild as HTMLElement).style?.fontSize;
    expect(fontSize).toBe('32px');
  });
});
