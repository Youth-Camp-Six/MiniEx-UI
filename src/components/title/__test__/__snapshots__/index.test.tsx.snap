// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title title snapshot test 1`] = `
<div>
  <h1
    class="mi-title"
    style="font-size: 14px;"
  >
    标题组件 s
  </h1>
  <h1
    class="mi-title"
    style="font-size: 18px;"
  >
    标题组件 l
  </h1>
  <h1
    class="mi-title"
    style="font-size: 16px;"
  >
    标题组件 m
  </h1>
  <h1
    class="mi-title"
    style="font-size: 24px;"
  >
    标题组件 xl
  </h1>
  <h1
    class="mi-title"
    style="font-size: 32px;"
  >
    标题组件 xxl
  </h1>
</div>
`;
